presto-main/src/main/java/io/prestosql/sql/analyzer/Analyzer.java

```diff
@@ -72,6 +75,7 @@ public Analyzer(
         this.parameters = parameters;
         this.parameterLookup = parameterLookup;
         this.warningCollector = requireNonNull(warningCollector, "warningCollector is null");
+        this.statsCalculator = statsCalculator;
```

Comment:
`rnn`

more importantly though -- it's counter intuitive why Analyzer needs stats calculator. It seems analysis result doesn't depend on stats (the plan does). I see this is needed to pass to ShowStatsRewrite. Maybe we can inject it there at creation?

Reply:
`Analyzer` is created on demand, most of it dependencies are session bound. Only `statsCalculator` is bound in guice. I can do what you suggest, but then I would need to pass `StatementRewrite` as parameter of `Analyzer` constructor. I can also extract `AnalyzerFactory`. Both options sounds to me like an overkill. WDYT?

Reply:
right. nvrm